### PR TITLE
Add little-endian and big-endian read functions for InputStreams

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:

**New feature** (*non-breaking change which adds functionality*)

## Description:

This PR adds thin overloads of `ReadValueLE`/`ReadValueBE` allowing to read data of different endianness directly from `InputStreams`. This will be needed for parsing TIFFs.

### Affected modules and functionalities:
`core/byte_io.h`

### Tests:

- [x] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

New tests are `byte_io_test`: `read_value_unsigned_int_input_stream`, `read_value_float_input_stream` which also check if the stream pos is moving forward on `ReadValueBE/LE`.

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2862

### CODE REVIEW

File: include/dali/core/byte_io.h [hunk 1]
```diff
@@ -41,6 +43,13 @@ void ReadValueImpl(float &value, const uint8_t* data) {
   memcpy(&value, &tmp, sizeof(float));
 }
 
+template <int nbytes, bool is_little_endian, typename T>
+void ReadValueImpl(T &value, InputStream &stream) {
+  uint8_t data[nbytes];
+  stream.ReadBytes(data, nbytes);
+  return ReadValueImpl<nbytes, is_little_endian>(value, data);
+}
+
```
Comment (jantonguirao) [inline include/dali/core/byte_io.h:52]: Maybe also add the 
```template <int nbytes, bool is_little_endian>
void ReadValueImpl(float &value, InputStream &stream) {```
version too, just like in the case of data ptr?

Comment (jantonguirao) [reply include/dali/core/byte_io.h:52]: Yeah, you are absolutely right. Sorry for the noise

**Review by jantonguirao — APPROVED** [review]

File: dali/core/byte_io_test.cc [hunk 2]
```diff
@@ -59,6 +60,28 @@ TEST(byte_io, read_value_float) {
   EXPECT_EQ(1.0f, ReadValueLE<float>(data_le));
 }
 
+TEST(byte_io, read_value_unsigned_int_input_stream) {
+  const uint8_t data[] = {0x04, 0xd2, 0x1e, 0x61};  // dec 7777 = hex 0x1E61
```
Comment (mzient) [inline dali/core/byte_io_test.cc:64]:
```suggestion
  const uint8_t data_be[] = {0x04, 0xd2, 0x1e, 0x61};  // dec 7777 = hex 0x1E61
```
x86, ARM (and most computers, in fact) natively work with LE, so marking data as "LE" and not marking "BE" as such is misleading.

Comment (szkarpinski) [reply dali/core/byte_io_test.cc]: Done

**Pull request merged**